sonic_platform_base/sonic_xcvr/xcvr_api_factory.py

```diff
@@ -0,0 +1,28 @@
+"""
+    xcvr_api_factory.py
+
+    Factory class responsible for instantiating the appropriate XcvrApi
+    implementation for a xcvr module in SONiC
+"""
+
+from .xcvr_eeprom import XcvrEeprom
+# TODO: remove the following imports
+from .codes.public.sff8024 import Sff8024
+from .api.public.cmis import CmisApi
+from .mem_maps.public.cmis import CmisMemMap
+
+class XcvrApiFactory(object):
```

Comment:
How is XcvrApiFactory linked with XcvrApi such that functions in class SfpOptoeBase could use functions in class XcvrApi?

Reply:
SfpOptoeBase inherits from SfpBase, which contains a reference to XcvrApiFactory. SfpBase uses this reference to instantiate XcvrApi, which it then exposes via get_xcvr_api().

Reply:
Yeah, I can see the logic here being SfpOptoeBase => SfpBase => XcvrApiFactory. But I did not see anywhere in XcvrApiFactory that Class XcvrApi has been imported. That was my original question. Hope this clarifies it.

Reply:
The XcvrApi class is meant to be an abstract base class and thus you would not see it explicitly imported in XcvrApiFactory. Right now the only imports in XcvrApiFactory are for concrete implementations of XcvrApi like CmisApi, and later Sff8436Api, Sff8472Api, etc., though this is for the interim since ideally we make use of the id_mapping feature described in the design to only import the classes that are needed. 

To clarify, SfpBase uses XcvrApiFactory to instantiate a XcvrApi-derived class (which is assumed to implement all of the methods). The get_xcvr_api() method is called by client code to fetch some object which follows the XcvrApi interface, but could be one of many different XcvrApi-based classes depending on the module detected.